File: typescript/src/models/contentType.ts
// this file is @generated

export enum ContentType {
  Plain = "plain",
  Markdown = "markdown",
}

export const ContentTypeSerializer = {
  _fromJsonObject(object: any): ContentType {
    return object;
  },

  _toJsonObject(self: ContentType): any {
    return self;
  },
};

File: rust/src/models/content_type.rs
// this file is @generated
use std::fmt;

use serde::{Deserialize, Serialize};

#[derive(Clone, Copy, Debug, Eq, PartialEq, Ord, PartialOrd, Hash, Serialize, Deserialize)]
pub enum ContentType {
    #[serde(rename = "plain")]
    Plain,
    #[serde(rename = "markdown")]
    Markdown,
}

impl fmt::Display for ContentType {
    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
        let value = match self {
            Self::Plain => "plain",
            Self::Markdown => "markdown",
        };
        f.write_str(value)
    }
}
